arm11/source/start.s

```diff
@@ -85,6 +85,8 @@ wait_arm9:
 	ldr r1, =0xFF
 	str r1, [r0, #SCU_INV_ALL_REG]
 
+	bl enable_cores23
```

Comment:
We really need to setup a sane stack pointer when we start using C (as well as manually clear .bss)

Reply:
https://github.com/profi200/open_agb_firm/blob/1fcf024520571c9e8732a5aaaf1bbd7661cba423/source/arm11/start.s#L111-L128 looks highly relevant.

Reply:
Yeah, we don't really need the heap (at least I hope we won't) but stacks are essential because the C runtime straight up assumes it's valid, we've had trouble in the past over that.

If we're only going to be in svc mode, simply allocating some static space in bss and then pointing so to the top of that space should be enough. If we want to handle interrupts (not in the polling way we do rn, but proper interrupts) a lot more setup is needed for the interrupt vectors and memory management in general.

Reply:
Note that the above code is pretty messy. There are certainly better ways to handle stacks and multiple cores.

Reply:
Pretty much the same as the code above:
```c
#define STACKS_BOTTOM   0x.... /* or coming from linker script */
#define STACK_SIZE_LOG2 12
```
```asm
mrc  p15, 0, r0, c0, c0, 5   @ Get CPU ID
ands r0, r0, #3
...
ldr sp, STACKS_BOTTOM
add sp, [r0, lsl #STACK_SIZE_LOG2] @ Stack bottom per core
add sp, #STACK_SIZE_LOG2           @ Stack top per core
```

Reply:
thankfully secondary cores dont need a stack, their code can all be written in asm for simplicity since it's just a spin table

Reply:
Done. PTAL.